rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/ExpandedGetResponseFilter.java

```diff
@@ -244,6 +262,26 @@ private void attachExpandedGetsBody(
     }
   }
 
+  private Object extractHeader(HttpRequest<?> request, Header header) {
+
+    if (ACCEPT_LANGUAGE.equals(header.getName())) {
+      Optional<Locale> locale = request.getLocale();
+
+      if (header.isMandatory()) {
+        return locale.orElseThrow(
+            () ->
+                new IllegalStateException(
+                    String.format(
+                        "The header '%s' is mandatory but was not found in the request",
+                        header.getName())));
+      }
+
+      return locale.orElse(null);
+    }
+
+    return request.getHeaders().get(header.getName());
```

Comment:
`equest.getHeaders().get(header.getName())` returns an `Optional` but above you return a real value such as the `Locale`. I don't think this is expected. I would return String values for the `else` case.

The mandatory checking should apply to other headers too.